### DISCUSSION TITLE
Import Live WordPress Site to Laragon

### DISCUSSION THREAD

**Discussion by dan-cloud** [leokhoa/laragon#673] [Q&A] [answered]
Good day! What are the best practices for creating a local environment for an existing WordPress site on Laragon? Some suggestions I've read from searches suggesting replacing local wp-content and the sql file with the respecitve from the live sight but it's not clear to me whether this will get me the desired result. I did notice that another community member backed up his production site and restored to a Laragon site.

**Comment by dustinatx** [comment, marked as answer]
This is what I do:

- Backup the entire WordPress directory and database from production and download those files.
- Create a new folder in www and unzip the files there.
- Create a new database with the same name as the folder. I use phpMyAdmin for this and latin1_swedish_ci Collation.
- Use the command line to import the database file into the database. (mysql -u username -p database_name < file.sql)
- Edit two options in wp_options with the local URL: siteurl & home
- Update wp-config.php with my local database info.
- Login to the site and use [Better Search Replace](https://wordpress.org/plugins/better-search-replace/) to update the URL throughout the database.

**Reply by aaronedev** [reply]
thanks dawg

**Reply by dan-cloud** [reply]
Thanks @dustinatx ! I was able to get my prod site running on my local. Am still challenged with an issue that I don't expect you to have an answer on. When I try to access WP-admin, I get a fatal error from the Elementor-pro plugin. Seems to have stopped and quit at licesnse\API.php. Anyway, I may contact that vendor directly if I can't figure it out.

**Reply by dustinatx** [reply]
Here is an idea:

Rename the elementor plugin folder. For example, elementor-temp. Try getting into wp-admin. I bet it loads because Elementor will be disabled. Then use Better Search Replace to update the URL throughout the database. Finally, rename the elementor directory back to its original state & reactivate elementor.

Another idea is to install wp-cli and use that to run a search and replace on the database.

https://gist.github.com/ladislavsulc/8b7450fa7cddd0129e0dcfbbbc5a6647

https://www.inmotionhosting.com/support/edu/wordpress/wp-cli/wp-cli-search-replace-command/#:~:text=WP%2DCLI's%20search%2Dreplace%20command,use%20https%3A%2F%2Fexample.com.

**Comment by dan-cloud** [comment]
I was finally able to get up and running on wp-admin. It turned out that the fatal error that I was seeing in Elementor Pro was due to one of two things. 1) A versioning issue between the Elementor Pro and PHP. I installed PHP 7.4 on Laragon as this was the version that my site was using in Production. 2) Elementor Pro has a license restriction that forces users to use only the specific domain name on their localhost to match that of the site to which the license is registered. So google.com would need to be created as google.test, google.dev, etc to avoid error.

@dustinatx Thank you much for all of the steps! I've added to my internal wiki.

**Reply by aaronedev** [reply]
dudeee the same for me thanks for sharing - wish i didnt use elementor back in the day when creating that website haha

**Comment by pacmanito** [comment]
Maybe this information can be useful for someone:
I made all aforementioned steps but main page was downloaded (other pages worked just fine) instead of being opened. The solution to the issue was to clear site cookies in Chrome and hit "clear site storage" in dev-tools > application. So if you have similar issue check if the page is opened in incognito.
FYI also I ran several times into issues with uploads not working due to old value of "upload_path" in wp_options.

**Comment by MichaelA2014** [comment]
Use updraft pro plugin. You install it on source and on target then sync them together using a secret key and you can push / pull data between two sites in both directions.
